### DISCUSSION TITLE
How to check wsdd status?

### DISCUSSION THREAD

**Discussion by gitowiec** [christgau/wsdd#192] [Q&A] [answered]
Hi!
I'm running OpenMediaVault as a server (it's version is based on Debian Bullseye). I already have wsdd installed (running apt install made it "set to manually installed"). I can run it via calling wsdd.py but everything I get is some Python error traces and "OSError: [Errno 98] Address already in use" message (it is printed twice, I assume because of two network adapters). Then there is no more output, it looks like program is running in the foreground. When I try to lookup my OMV server in Windows 10 Network Neighborhood, it is missing (the SAMBA/CIFS is configured correctly, I can access my server via \\\ip.address.of.omv\ in Explorer). So my conclusion is that wsdd is not running. Or is it? How can I check the status of wsdd on my server?

<img width="731" alt="Zrzut ekranu 2024-01-03 184002" src="https://github.com/christgau/wsdd/assets/1553195/328aa2e5-b516-4db0-aac9-11703e22109c">
<img width="680" alt="Zrzut ekranu 2024-01-03 184044" src="https://github.com/christgau/wsdd/assets/1553195/7531e119-9b84-4659-b7e0-3a992f04892c">

**Comment by christgau** [comment, marked as answer]
>  I already have wsdd installed (running apt install made it "set to manually installed")

Could be that a daemon was installed (and started) by doing so. 

I don't use or know OMV, but since it is based on Debian you could check if a service is running using `systemctl status wsdd`. You can also run `ss -tlnp | grep 5357` with root privileges. If there is non-empty output, an instance of wsdd (or another implementation) is already running. This would also explain the error messages ("address already in use") you see. 

> Then there is no more output, it looks like program is running in the foreground. 

Yes, because it does not daemonize itself.

> When I try to lookup my OMV server in Windows 10 Network Neighborhood, it is missing (the SAMBA/CIFS is configured correctly, 

This could also be caused by strict firewall setup. 

I would recommend to check if a wsdd daemon is running (see above) and stop it. After that run wsdd on the command line (just as you did). That should not cause error messages at all. When wsdd starts up successfully and firewall rules allow for it, your machine should show up immediately in Explorer's network view. 

Does this help?

**Reply by gitowiec** [reply]
Thank you, as you were writing this answer I just found how to manage services and read their logs on Debian. Turning off Windows 10 firewall just for private networks made it work (and of course restarting Windows box).

**Reply by christgau** [reply]
> Thank you, as you were writing this answer I just found how to manage services and read their logs on Debian.

Alright.

>  Turning off Windows 10 firewall just for private networks made it work (and of course restarting Windows box).

Generally, you should not have to disable the firewall on the Windows machine. Wsdd works fine with the Windows firewall being up and respective ports open, but that's managed by Windows itself. Sometimes the service discovery daemon (aka "services") on Windows have to be enabled explicitly.
